src/daft-functions/src/list/unique.rs

```diff
@@ -0,0 +1,160 @@
+                    Series::try_from_field_and_arrow_array(
+                        Field::new(input.name(), DataType::List(inner_type.clone())),
+                        list_array,
+                    )?
+                } else {
+                    input.clone()
+                };
+
+                let list = input.list()?;
+                let mut offsets = Vec::new();
+                offsets.push(0i64);
+                let mut current_offset = 0i64;
+                let mut result = Vec::new();
+
+                for sub_series in list {
+                    if let Some(sub_series) = sub_series {
+                        let probe_table = if self.ignore_nulls {
+                            sub_series.build_probe_table_without_nulls()?
+                        } else {
+                            sub_series.build_probe_table_with_nulls()?
+                        };
+
+                        let mut indices: Vec<_> = probe_table.keys().map(|k| k.idx).collect();
+                        indices.sort_unstable();
+
+                        let mut unique_values = Vec::new();
+                        for idx in indices {
+                            unique_values.push(sub_series.slice(idx as usize, (idx + 1) as usize)?);
+                        }
+
+                        current_offset += unique_values.len() as i64;
+                        offsets.push(current_offset);
+                        result.extend(unique_values);
+                    } else {
+                        offsets.push(current_offset);
+                    }
+                }
+
+                let field = Arc::new(input.field().to_exploded_field()?);
+                let child_data_type = if let DataType::List(inner_type) = input.data_type() {
+                    inner_type.as_ref().clone()
+                } else {
+                    return Err(DaftError::TypeError("Expected list type".into()));
+                };
+
+                if current_offset == 0 {
+                    let empty_array = arrow2::array::new_empty_array(child_data_type.to_arrow()?);
+                    let list_array = ListArray::new(
+                        Arc::new(Field::new(input.name(), input.data_type().clone())),
+                        Series::from_arrow(field, empty_array)?,
+                        OffsetsBuffer::try_from(offsets)?,
+                        input.validity().cloned(),
+                    );
+                    return Ok(list_array.into_series());
+                }
+
+                let result_refs: Vec<&Series> = result.iter().collect();
+                let mut growable = make_growable(
+                    &field.name,
+                    &child_data_type,
+                    result_refs,
+                    false,
+                    current_offset as usize,
+                );
+
+                for (i, series) in result.iter().enumerate() {
+                    growable.extend(i, 0, series.len());
+                }
```

Comment:
The way you're using the growable here still requires two copies of the data, once to construct `result` and once at `growable.build()`. Happy to chat about how you could do this more efficiently!